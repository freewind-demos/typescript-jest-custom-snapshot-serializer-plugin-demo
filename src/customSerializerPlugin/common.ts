export const KEY = Symbol.for('RawJsonWrapper');

export type RawJsonWrapper = {
  [KEY]: any
}
