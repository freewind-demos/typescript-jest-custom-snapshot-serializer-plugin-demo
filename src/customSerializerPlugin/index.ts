export * from './rawJsonWrapper';
export * from './MySerializerPlugin';
