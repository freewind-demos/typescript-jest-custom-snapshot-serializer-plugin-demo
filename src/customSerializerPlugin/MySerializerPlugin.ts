import {Config, NewPlugin, Refs} from 'pretty-format';
import {Printer} from 'pretty-format/build/types';
import {KEY, RawJsonWrapper} from './common';

function isRawJsonWrapper(data: any): data is RawJsonWrapper {
  return data.hasOwnProperty(KEY);
}

function unwrap(data: any): any {
  return isRawJsonWrapper(data) ? data[KEY] : undefined;
}

export const rawJsonSerializerPlugin: NewPlugin = {
  serialize(val: any, config: Config, indentation: string, depth: number, refs: Refs, printer: Printer): string {
    return JSON.stringify(unwrap(val), null, 2);
  },
  test(val: any): boolean {
    return isRawJsonWrapper(val);
  }
}
