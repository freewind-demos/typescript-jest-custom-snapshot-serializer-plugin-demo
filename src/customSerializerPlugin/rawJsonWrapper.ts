import {KEY, RawJsonWrapper} from './common';

export function rawJson(data: any): RawJsonWrapper {
  return {
    [KEY]: data
  }
}
