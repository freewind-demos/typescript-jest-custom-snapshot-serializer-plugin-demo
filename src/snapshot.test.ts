import {rawJson} from './customSerializerPlugin';

const data = {
  aaa: 111,
  array: [1, 2, 3]
}

describe('test', () => {
  it('snapshot normal', () => {
    expect(data).toMatchSnapshot();
  })
  it('snapshot rawJson', () => {
    expect(rawJson(data)).toMatchSnapshot();
  })
})
