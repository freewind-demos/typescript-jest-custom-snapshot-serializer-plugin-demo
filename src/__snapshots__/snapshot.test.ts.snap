// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot normal 1`] = `
Object {
  "aaa": 111,
  "array": Array [
    1,
    2,
    3,
  ],
}
`;

exports[`test snapshot rawJson 1`] = `
{
  "aaa": 111,
  "array": [
    1,
    2,
    3
  ]
}
`;
